gradle-witchcraft-logging/build.gradle

```diff
@@ -0,0 +1,61 @@
+apply plugin: 'groovy'
+apply plugin: 'java-gradle-plugin'
+apply plugin: 'com.gradle.plugin-publish'
+apply plugin: 'com.palantir.external-publish-jar'
+
+dependencies {
+    api gradleApi()
+    implementation project(':witchcraft-logging-formatting')
+    implementation 'com.google.guava:guava'
+
+    testImplementation gradleTestKit()
+    testImplementation 'com.google.guava:guava'
+    testImplementation 'com.netflix.nebula:nebula-test'
+    testImplementation 'org.assertj:assertj-core'
+    testImplementation 'org.mockito:mockito-core'
+    testImplementation 'org.mockito:mockito-junit-jupiter'
+    testImplementation 'org.junit.jupiter:junit-jupiter'
+
+    testRuntimeOnly 'org.junit.vintage:junit-vintage-engine'
+}
+
+gradlePlugin {
+    plugins {
+        witchcraftLoggingPlugin {
+            id = 'com.palantir.witchcraft-logging'
+            implementationClass = 'com.palantir.witchcraft.java.logging.gradle.WitchcraftLoggingPlugin'
+        }
+        witchcraftLoggingIdeaPlugin {
+            id = 'com.palantir.witchcraft-logging-idea'
+            implementationClass = 'com.palantir.witchcraft.java.logging.gradle.idea.IdeaPluginRecommenderPlugin'
+        }
+        witchcraftLoggingTestReportPlugin {
+            id = 'com.palantir.witchcraft-logging-testreport'
+            implementationClass = 'com.palantir.witchcraft.java.logging.gradle.testreport.TestReportFormattingPlugin'
+        }
+    }
+}
+
+pluginBundle {
+    website = 'https://github.com/palantir/witchcraft-java-logging'
+    vcsUrl = 'https://github.com/palantir/witchcraft-java-logging'
+    description = 'Palantir Witchcraft logging plugins used to improve developer experience building, running, and debugging applications using the witchcraft logging api: https://github.com/palantir/witchcraft-api.'
+    tags = ['witchcraft', 'logging', 'idea']
+
+    plugins {
+        witchcraftLoggingPlugin {
+            displayName = 'Palantir Witchcraft Logging Plugin'
+        }
+        witchcraftLoggingIdeaPlugin {
+            displayName = 'Palantir Witchcraft Logging Idea Plugin'
+        }
+        witchcraftLoggingTestReportPlugin {
+            displayName = 'Palantir Witchcraft Logging Test Report Rendering Plugin'
+        }
+    }
+}
+
+tasks.publish.dependsOn tasks.publishPlugins
+publishPlugins.onlyIf { versionDetails().isCleanTag }
+project.ext.'gradle.publish.key' = System.env["GRADLE_KEY"]
+project.ext.'gradle.publish.secret' = System.env["GRADLE_SECRET"]
```

Comment:
https://github.com/palantir/gradle-external-publish-plugin/blob/c9b0a813686b3a5703f5fa41a091858b448eddea/src/main/java/com/palantir/gradle/externalpublish/ExternalPublishGradlePluginPlugin.java#L26

For some reason this isn't in the readme.

Reply:
I borrowed heavily from baseline.

re: readme -- I'd forgotten to add a readme section to this PR describing the gradle plugin. I'll put something together shortly.

Reply:
Updated to use the plugin, good call on that. I'll pull it into gradle-baseline as well :-)